File: 0283-move-zeroes/0283-move-zeroes.java
/**
    move all non-zeros first and then append all zeros
    tc - o(n)
    sc - o(1)
 */
class Solution {
    public void moveZeroes(int[] nums) {
        int idx=0,n=nums.length;
        for(int i=0;i<n;i++){
            if(nums[i] != 0){
                nums[idx] = nums[i];
                idx++;
            }
        }

        for(int i=idx;i<n;i++){
            nums[i] = 0;
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let idx=0,i=0;

    while(i<nums.length){
        if(nums[i] != 0){
            nums[idx] = nums[i];
            idx++;
        }
        i++;
    }

    while(idx<nums.length){
        nums[idx]=0;
        idx++;
    }
};